From a9f3abdbb449f916ff150c83c4ed79d8ff1f09eb Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:36 -0400
Subject: [fs] ext4: fix over-defensive complaint after journal abort

Message-id: <1414600485-1929-47-git-send-email-lczerner@redhat.com>
Patchwork-id: 98548
O-Subject: [RHEL 7.1 PATCH v2 46/55] ext4: fix over-defensive complaint after journal abort
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Dmitry Monakhov <dmonakhov@openvz.org>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit c5d311926da483951bd5da637ed65de8614d1901

 Reviewed-by: Jan Kara <jack@suse.cz>
 Signed-off-by: Dmitry Monakhov <dmonakhov@openvz.org>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/ext4_jbd2.c b/fs/ext4/ext4_jbd2.c
index 0074e0d..3445035 100644
--- a/fs/ext4/ext4_jbd2.c
+++ b/fs/ext4/ext4_jbd2.c
@@ -256,8 +256,8 @@ int __ext4_handle_dirty_metadata(const char *where, unsigned int line,
  set_buffer_prio(bh);
  if (ext4_handle_valid(handle)) {
   err = jbd2_journal_dirty_metadata(handle, bh);
-  /* Errors can only happen if there is a bug */
-  if (WARN_ON_ONCE(err)) {
+  /* Errors can only happen due to aborted journal or a nasty bug */
+  if (!is_handle_aborted(handle) && WARN_ON_ONCE(err)) {
    ext4_journal_abort_handle(where, line, __func__, bh,
         handle, err);
    if (inode == NULL) {
-- 
1.7.1